processor/transformprocessor/internal/common/processor.go

```diff
@@ -30,8 +31,7 @@ func (r resourceStatements) Context() ContextID {
 }
 
 func (r resourceStatements) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
-	for i := 0; i < td.ResourceSpans().Len(); i++ {
-		rspans := td.ResourceSpans().At(i)
+	for _, rspans := range td.ResourceSpans().All() {
```

Comment:
IMHO, as you're working on profiles, you should keep your changes constrained to profiles. These cosmetic changes can happen in another PR.

Reply:
Pretty sure that one of the reviewers (here or in another PR) asked me to change this. Keeping this for now, until it becomes a blocker for someone.